src/main/java/stringcalculator/StringCalculator.java

```diff
@@ -0,0 +1,51 @@
+package stringcalculator;
+
+import java.util.Arrays;
+
+public class StringCalculator {
+
+
+
+    public StringCalculator() {
+    }
+
+    public int calculate(String str){
+
+        if(str == null || str.equals("")){
+            return 0;
+        }
+
+        if(isDigit(str)){
+            return Integer.valueOf(str);
+        }
+
+
+        String[] split = str.split(getDelimiter(str));
+
+        if(!Arrays.stream(split).allMatch(this::isDigit)){
```

Comment:
allMatch 메서드는 stream 의 모든 인자를 체크하는 것으로 알고 있습니다.
이 문장은 stream 의 요소 중 isDigit 을 만족하지 않는 요소가 하나라도 있으면 실행되는 조건으로 보이는데요!
```suggestion
        if(Arrays.stream(split).anyMatch(token -> !this.isDigit(token))){
```
이처럼 반대로 생각해서 검증횟수를 줄여보는 것은 어떨까요!
(제가 반대로 잘 생각한게 맞겠죠?😅)